docs/specs/build/toc.yml

```diff
@@ -391,3 +376,156 @@ inputs:
 outputs:
   build.log: |
     ["error","circular-reference","Found circular reference: docs/toc.md --> docs/toc.md","docs/toc.md"]
+---
+# Reference to a folder1
+inputs:
+  docfx.yml:
+  docs/toc.md: |
+    # [Reference TOC File 1](f1/toc.yml)
+  docs/f1/toc.yml: |
+    - name: Toc Reference
+      href: f11/toc.md
+    - name: Folder Reference1
+      href: f11/
+    - name: Folder Reference2
+      href: f11/
+      topicHref: f11/a.md
+  docs/f1/f11/toc.md: |
+    # [Index Reference](index.md)
+  docs/f1/f11/index.md:
+  docs/f1/f11/a.md:
+outputs:
+  docs/f1/f11/a.json:
+  docs/f1/f11/index.json:
+  docs/toc.json: |
+   {  
+       "items":[  
+          {  
+             "children":[  
+                {  
+                   "children":[  
+                      {  
+                         "href":"f1/f11/",
+                         "toc_title":"Index Reference"
+                      }
+                   ],
+                   "toc_title":"Toc Reference"
+                },
+                {  
+                   "href":"f1/f11/",
+                   "toc_title":"Folder Reference1"
+                },
+                {  
+                   "href":"f1/f11/a",
+                   "toc_title":"Folder Reference2"
+                }
+             ],
+             "toc_title":"Reference TOC File 1"
+          }
+       ]
+    }
+  build.manifest:
+---
+# Reference to a folder2
+# Folder referenced toc doesn't belong to nested toc
+inputs:
+  docfx.yml:
+  docs/toc.md: |
+    # [Folder Reference](f1/)
+  docs/f1/toc.md: |
+    # [Index Reference](index.md)
+  docs/f1/index.md:
+outputs:
+  docs/f1/index.json:
+  docs/toc.json: |
+   {  
+       "items":[  
+          {  
+             "href": "f1/",
+             "toc_title":"Folder Reference"
+          }
+       ]
+    }
+  docs/f1/toc.json: |
+   {  
+       "items":[  
+          {  
+             "href": ".",
+             "toc_title":"Index Reference"
+          }
+       ]
+    }
+  build.manifest:
+---
+# Reference to a folder3
+# The href should be the first child with href in referenced toc
+inputs:
+  docfx.yml:
+  docs/toc.md: |
+    # [Folder Reference1](f1/)
+    # [Folder Reference2](f2/)
+  docs/f1/toc.md: |
+    # Title
+    # [File Reference](b.md)
+  docs/f2/toc.md: |
+    # Title1
+    ## Child Title1
+    ## Child Title2
+    # Title2
+    ## [File Reference](a.md)
+  docs/f1/b.md:
+  docs/f2/a.md:
+outputs:
+  docs/f2/a.json:
+  docs/f1/b.json:
+  docs/toc.json: |
+   {  
+       "items":[  
+          {  
+             "href":"f1/b",
+             "toc_title":"Folder Reference1"
+          },
+          {  
+             "href":"f2/a",
+             "toc_title":"Folder Reference2"
+          }
+       ]
+    }
+  docs/f1/toc.json: |
```

Comment:
can you simplify test specs to help us easily understand what is been tested:
- remove duplicated inputs
- condense output to only included what is been verified

```yamlinputs:
# href points to the first article in referenced toc
inputs:
  docfx.yml:
  docs/toc.md: |
    # [a](dir/)
  docs/dir/toc.md: |
    # Title
    ## Child Title
    # [First Article](a.md)
  docs/dir/a.md:
outputs:
  docs/toc.json: |
   { "items": [{ "toc_title":"a", "href":"dir/a" }] }
  docs/dir/toc.json:
  docs/dir/a.json:
  build.manifest:
```